package NYT.NProto;

import "yt/core/yson/proto/protobuf_interop.proto";
import "yt/core/ytree/proto/attributes.proto";

enum EColor
{
    Color_Green = -1 [(NYT.NYson.NProto.enum_value_name) = "green"];
    Color_Red = 2 [(NYT.NYson.NProto.enum_value_name) = "red"];
    Color_Blue = 3 [(NYT.NYson.NProto.enum_value_name) = "blue"];
}

message TNestedMessage
{
    optional int32 int32_field = 1;
    optional EColor color = 19;
    optional TNestedMessage nested_message = 2;
    repeated int32 repeated_int32_field = 100;
}

message TMessage
{
    optional int32 int32_field_xxx = 1 [(NYT.NYson.NProto.field_name) = "int32_field"];
    optional uint32 uint32_field = 2;
    optional sint32 sint32_field = 3;
    optional int64 int64_field = 4;
    optional uint64 uint64_field = 5;
    optional sint64 sint64_field = 6;
    optional fixed32 fixed32_field = 7;
    optional fixed64 fixed64_field = 8;
    optional sfixed32 sfixed32_field = 9;
    optional sfixed64 sfixed64_field = 10;
    optional bool bool_field = 11;
    optional string string_field = 12;
    optional float float_field = 13;
    optional double double_field = 14;
    optional TNestedMessage nested_message1 = 15;
    optional TNestedMessage nested_message2 = 16;
    repeated int32 repeated_int32_field = 17;
    repeated TNestedMessage repeated_nested_message1 = 18;
    repeated TNestedMessage repeated_nested_message2 = 21;
    optional NYT.NYTree.NProto.TAttributeDictionary attributes = 19;
    optional bytes yson_field = 20 [(NYT.NYson.NProto.yson_string) = true];
}

message TNestedMessageWithRequriedFields
{
    optional int32 optional_field = 1;
    required int32 required_field = 2;
}

message TMessageWithRequiredFields
{
    optional int32 optional_field = 1;
    required int32 required_field = 2;
    repeated TNestedMessageWithRequriedFields nested_messages = 3;
}
